frontend/src/components/utils/autoRefreshHandler.tsx

```diff
@@ -31,29 +45,92 @@ export default (
 
     componentDidUpdate(preProps: any) {
       if (this.props !== preProps) {
-        this.setState({ items: [] }, this.fetchInfo);
+        this.setState({
+          items: null,
+          itemsLength: config.count,
+          display: false,
+          hasMore: true
+        });
       }
     }
 
-    fetchInfo = () => {
-      fetchDataFn()
+    fetchInfo = (count: number) => {
+      config
+        .fetchDataFn(count)
         .then((items: any) => {
-          this.setState({ items });
+          if (items.length > 0) {
+            this.setState({ items, display: true });
+          } else {
+            this.setState({ hasMore: false, display: true });
+          }
         })
         .catch((err: any) => {
           console.error(err);
         });
     };
 
     regularFetchInfo = () => {
-      this.fetchInfo();
+      this.fetchInfo(this.state.itemsLength);
       if (this.timer !== null) {
         this.timer = setTimeout(this.regularFetchInfo, 10000);
       }
     };
 
+    fetchMoreData = async () => {
+      if (config.dashboard) {
+        this.setState({ hasMore: false });
+        return;
+      }
+      if (this.state.items.length > 0) {
+        const endTimestamp = this.getEndTimestamp(config.categary);
+        const newData = await config.fetchDataFn(config.count, endTimestamp);
+        if (newData.length > 0) {
+          const items = this.state.items.concat(newData);
+          this.setState({ items, itemsLength: items.length });
+        } else {
+          this.setState({ hasMore: false });
+        }
+        return;
+      }
+    };
+
+    getEndTimestamp = (categary: string) => {
+      let endTimestamp;
+      switch (categary) {
+        case "Account":
+          endTimestamp = this.state.items[this.state.items.length - 1]
+            .createdAtBlockTimestamp;
+          break;
+        case "Block":
+          endTimestamp = this.state.items[this.state.items.length - 1]
+            .timestamp;
+          break;
+        case "Node":
+          endTimestamp = this.state.items[this.state.items.length - 1].lastSeen;
+          break;
+        case "Transaction":
+          endTimestamp = this.state.items[this.state.items.length - 1]
+            .blockTimestamp;
+          break;
+        default:
+          endTimestamp = undefined;
```

Comment:
There are at least three options to put the responsibility for this logic to:

1. (this way) put it into the downline component and teach it to handle every single possible type of data. The problem is that you have to hard-code every single usage scenario (it is not scalable and fragile).
2. (previous implementation) enforce the external contract and require the data to be of a specific shape (having "timestamp" field in every item). This works, but you have to repack the data and in the previous attempt you had to go throughout the codebase and rename the field that makes sense for the specific type into a generic "timestamp". As you can see, the change is intrusive to the whole codebase, and it is also fragile and creates a tight coupling from the API through the relevant components down to the autoRefreshHandler.
3. (what I suggested) design authRefreshHandler to receive a function (similarly to `fetchDataFn`) which will receive an item and return a range indicator ("timestamp"), this way autoRefreshHandler won't need to know anything about the structure of the `item`, and there is no need to change the APIs fetching, so all you need is to pass this "timestamp extractor" function to autoRefreshHandler.